MacOS integration

DantePuglisi opened the pull request:
A MacOS demo app target has been added to the project so the library can be used on any device with MacOS

DantePuglisi committed:
Added MacOS demo app

harrisonf-ledger reviewed (changes requested):
- Can we show an error if proper permissions are not granted?
- In the iOS target, the development team needs to be set to Ledger SAS
- Starscream is still listed as a Swift package dependency - can that be removed now?

DantePuglisi commented:
@harrisonf-ledger 

> * Can we show an error if proper permissions are not granted?
> * In the iOS target, the development team needs to be set to Ledger SAS
> * Starscream is still listed as a Swift package dependency - can that be removed now?

- Added error if permissions are not granted
- Changed iOS target's signing Team
- Starscream is a dependency of the demo project (used to install the BTC app), it shouldn't be included when adding `BleTransport` as a dependency

DantePuglisi committed:
Actioned requested changes
